src/ecdh.rs

```diff
@@ -86,6 +87,22 @@ impl Deref for SharedSecret {
 }
 
 
+unsafe extern "C" fn hash_callback<F>(output: *mut c_uchar, x: *const c_uchar, y: *const c_uchar, data: *mut c_void) -> c_int
+    where F: FnMut([u8; 32], [u8; 32]) -> SharedSecret {
+    let callback: &mut F = &mut *(data as *mut F);
+
+    let mut x_arr = [0; 32];
+    let mut y_arr = [0; 32];
+    ptr::copy_nonoverlapping(x, x_arr.as_mut_ptr(), 32);
+    ptr::copy_nonoverlapping(y, y_arr.as_mut_ptr(), 32);
```

Comment:
will it be a good idea to use slices instead of arrays and https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html ?

Reply:
Why?
don't you prefer the type to be an explicit compile time requirement?

Reply:
I think arrays are a bit nicer because we're passing them into a function, rather than making the user construct them (which would be unergonomic)